### DISCUSSION TITLE
Invalid args for response field!

### DISCUSSION THREAD

**Discussion by kenjoel** [fastapi/fastapi#6591] [Questions]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
class PyObjectId(ObjectId):
    @classmethod
    def __get_validators__(cls):
        yield cls.validate

    @classmethod
    def validate(cls, v):
        if not ObjectId.is_valid(v):
            raise ValueError("Invalid objectid")
        return ObjectId(v)

    @classmethod
    def __modify_schema__(cls, field_schema):
        field_schema.update(type="string")


class Item(BaseModel):
    id: PyObjectId = Field(default_factory=PyObjectId, alias="_id")
    name: str = None
    quantity: str = None
    price: int = None
    available: bool = None

    class Config:
        BaseConfig.arbitrary_types_allowed = True
        allow_population_by_field_name = True
        json_encoders = {ObjectId: str}



def all_items():
    expected_items = db["items"].findAll()
    return expected_items


app = FastAPI()
BaseConfig.arbitrary_types_allowed = True

@app.get("get_item", response_model=Item)
async def get_all_items():
    data = await get_all_items()
    return data
```

### Description

Guys, I have really tried to find a solution to this problem and I have done all that was recommended in github issues to no avail. I am still getting the same error. Where am I going wrong?

### Operating System

Linux

### FastAPI Version

0.68.1

### Python Version

3.9

### Additional Context

Please help

**Comment by Robin-Bonnin** [comment]
Hello! 

I haven't tested but it seems that you need to wait for the DB to get the response.
eg

    
      def all_items():
          return await db["items"].findAll()


There is a very good snippet on how to integrate Mongo: https://github.com/mongodb-developer/mongodb-with-fastapi/blob/master/app.py

**Comment by kenjoel** [comment]
Hi, I am doing so on this method 

@app.get("get_item", response_model=Item)
async def get_all_items():
    data = await get_all_items()
    return data



but my worry is the Error, its the same error with Pydantic not allowing arbitrary classes

"invalid args for response field! Hint: check that default=<class 'backend.models.Item.Item'> extra={} is a valid pydantic field type"

**Comment by Robin-Bonnin** [comment]
Hi! 

What I meant is more to do something like that
```python
class PyObjectId(ObjectId):
  @classmethod
  def __get_validators__(cls):
      yield cls.validate

  @classmethod
  def validate(cls, v):
      if not ObjectId.is_valid(v):
          raise ValueError("Invalid objectid")
      return ObjectId(v)

  @classmethod
  def __modify_schema__(cls, field_schema):
      field_schema.update(type="string")


class Item(BaseModel):
  id: PyObjectId = Field(default_factory=PyObjectId, alias="_id")
  name: str = None
  quantity: str = None
  price: int = None
  available: bool = None

  class Config:
      BaseConfig.arbitrary_types_allowed = True
      allow_population_by_field_name = True
      json_encoders = {ObjectId: str}



async def all_items(): # Add async here
  expected_items = await db["items"].findAll() # Add await here
  return expected_items


app = FastAPI()
BaseConfig.arbitrary_types_allowed = True

@app.get("get_item", response_model=Item)
async def get_all_items():
  data = await all_items() # Change the function called to the actual one that returns DB data
  return data
```

1. You are calling your endpoint function within your endpoint instead of the function that should get the data.
2. The function that gets the data is not async -> You cannot await
3. The function that get items needs to wait for the db to reply

By adding the model as response model, FastAPI will automatically serialize it to the correct model
Let me know if that works :)

**Comment by kenjoel** [comment]
This worked, Thank you so much for your help @Robin-Bonnin
